rotkehlchen/chain/optimism/proxies_inquirer.py

```diff
@@ -0,0 +1,30 @@
+from typing import TYPE_CHECKING, Optional
+from rotkehlchen.chain.evm.proxies_inquirer import EvmProxiesInquirer
+from rotkehlchen.types import ChecksumEvmAddress
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.optimism.node_inquirer import OptimismInquirer
+
+
+class OptimismProxiesInquirer(EvmProxiesInquirer):
+    """Class to retrive information about DSProxy for defi addresses."""
+
+    def __init__(
+            self,
+            optimism_inquirer: 'OptimismInquirer',
+    ) -> None:
+        super().__init__(optimism_inquirer)
+        self.optimism = optimism_inquirer
+        self.dsproxy_registry = self.optimism.contracts.contract('DS_PROXY_REGISTRY')
```

Comment:
if we have the logic already for the DSProxy why not move the common logic of the methods to the evm proxies inquierer? I beleive you just need to rename the self.ethereum and self.optimism to something like self.inquierer and then the logic in the methods should be common right?

Reply:
The thing is that I have not checked yet and not sure whether the contracts will be identical everywhere. Maybe they will have some little differences between chains? I just thought that is better to check these optimism-specific things in a different PR, because it is not mandatory for the issues that this PR solves. Does it make sense?

Reply:
I wouldn't leave incomplete things. Just checking if it is the same and adding a test for it shouldn't be much work. I wouldn't leave tasks for other PR since you are doing this change in this PR. Later we would need to move almost all the code related to this new class

Reply:
Okay